testsuite/features/step_definitions/command_steps.rb

```diff
@@ -1518,6 +1518,9 @@
 
   # Add the new hostname on controller's /etc/hosts to resolve in smoke tests
   `echo '#{server_node.public_ip} #{new_hostname}#{server_node.full_hostname.delete_prefix(server_node.hostname)} #{new_hostname}' >> /etc/hosts`
+
+  hostname = get_target('server').hostname
```

Comment:
Unnecessary variable assignment imho.
You can directly use the attribute in your comparison.
```
raise 'Wrong server hostname after changing it' unless get_target('server').hostname == new_hostname
```

Reply:
Thank you. This was only a test from my side. I am not 100% sure, yet, if I will leave that in.